File: exemplos/python/09_escopo_negado.py
# Prova de que uma chave de leitura NÃO escreve: POST /api/itens exige o escopo admin:inquilino
from _apoio import chamar, exigir, exigir_chave

exigir_chave()
status, _cab, corpo = chamar("/api/itens", metodo="POST", corpo={"tipo": "pasta", "titulo": "não deve ser criado"})
exigir(status == 403, f"esperava 403 (escopo insuficiente) e veio {status}: {corpo}")
exigir(corpo.get("erro") == "escopo_insuficiente", f"esperava erro escopo_insuficiente, veio {corpo.get('erro')}")
print(f"recusado como se espera: {corpo['mensagem']}")
print(f"exigido: {corpo['detalhe']['exigido']}; a chave tem: {', '.join(corpo['detalhe']['token_tem'])}")

File: exemplos/js/09_escopo_negado.mjs
// Prova de que uma chave de leitura NÃO escreve: POST /api/itens exige o escopo admin:inquilino
import { chamar, exigir, exigirChave } from "./_apoio.mjs";

exigirChave();
const { status, objeto } = await chamar("/api/itens", {
  metodo: "POST",
  corpo: { tipo: "pasta", titulo: "não deve ser criado" },
});
exigir(status === 403, `esperava 403 (escopo insuficiente) e veio ${status}: ${JSON.stringify(objeto)}`);
exigir(objeto.erro === "escopo_insuficiente", `esperava erro escopo_insuficiente, veio ${objeto.erro}`);
console.log(`recusado como se espera: ${objeto.mensagem}`);
console.log(`exigido: ${objeto.detalhe.exigido}; a chave tem: ${objeto.detalhe.token_tem.join(", ")}`);
